xAPI-Data.md

```diff
@@ -1608,70 +1608,17 @@ The asserting authority represents the authenticating user or some system or app
 
 ###### <a name="2.4.9.s3"></a>Requirements
 
-* <a name="2.4.9.s3.b1"></a>Authority MUST be an Agent, except in 3-legged OAuth, where it MUST be a Group with two Agents. 
-The two Agents represent an application and user together.
-* <a name="2.4.9.s3.b2"></a>The LRS MUST include the user as an Agent as the entire authority if a user connects 
-directly (using HTTP Basic Authentication) or is included as part of a Group. 
-* <a name="2.4.9.s3.b3"></a>The LRS MUST ensure that all Statements stored have an authority.
-* <a name="2.4.9.s3.b4"></a>The LRS SHOULD overwrite the authority on all Statements it stores,
+* <a name="2.4.9.s3.b1"></a>The LRS MUST ensure that all Statements stored have an authority.
+* <a name="2.4.9.s3.b2"></a>The LRS SHOULD overwrite the authority on all Statements it stores,
 based on the credentials used to send those Statements.
+* <a name="2.4.9.s3.b2"></a>When the LRS overwrites the authority, the LRS MUST apply a deterministic process to map
+the credentials used to store a statement to an authority.
```

Comment:
I thought this requirement was basically necessary in order to test anything to do with authority at all, though in a way it's insufficient because we don't (and can't reasonably due to the possibility of two completely different tokens representing the same principal) require different credentials to result in different authorities.

This may just be a case of "we don't need to specify the LRS shouldn't be stupid", not sure

Reply:
+1 for "we don't need to specify the LRS shouldn't be stupid", but if we don't already have it some text guidance what to set the authority to could be helpful. I'd like to explicitly discourage using the email address/identifier of the user who created the credentials/link as the authority, but maybe that's a best practice rather than something for the spec.

Reply:
The reason for not using an identifier of the user who creates the link, it is can lead to the scenario where the administrator of the account is the authority for everything, which can make it hard to distinguish between data sources (which is the only practical use of the authority property I've seen).

Reply:
In that case though it seems like it _is_ the user who creates the link who logically is the authority (they are authorizing the LRS to accept the statements). I can see where it would be useful to use some other information, but I'm hesitant to suggest it as a best practice. In any case, I'd rather leave that out of the spec so we have more flexibility around recommending the current best practice w/o going through a spec (and indeed eventually a standard) change.

Reply:
Isn't "authorizing the LRS to accept statements from <some source>" a separate kind of assertion from "the LRS should recognize <some source> as the authority for the statement(s) made"?

My understanding of @garemoko's point was not that the authority should be recognized as the agent that _made possible the integration_ between statement producer and consumer, but rather that the authority should be recognized as the _expresser/validator of the statement_... maybe I'm not understanding on this point?

Reply:
Yes. I think we get the most practical benefit when the authority represents the system or person using the credentials, rather than the system or person who created them.

Reply:
> the system or person using the credentials, rather than the system or person who created them
I agree with this. So in the statement forwarding case, you'd want to create credentials for the statement forwarder (because the LRS will have to set the authority based on the credentials)... I think no one is suggesting different authorities for one credential used?

> Isn't "authorizing the LRS to accept statements from {some source}" a separate kind of assertion from "the LRS should recognize {some source} as the authority for the statement(s) made"?

Yes, though the LRS has no way to know what the authority of the statements made should be except for the credentials used to connect and send those statements. Signed statements would be an exception, but even there it probably makes sense to just ignore the authority and look at the signature (when reading the statement, not for setting authority)

> but if we don't already have it some text guidance what to set the authority to could be helpful

Might this be the sort of thing that should go in a security best practice guide?

Perhaps this requirement should be left at the LRS is responsible for setting the authority? Is it OK to have the LRS set different authorities for the same credential based on some other factor?

Reply:
For example, if an LRS sets the authority to a new random UUID for each statement regardless of credentials, should it be breaking a requirement? Or is that just "not the best practice"